transformer/external/executabletransformer.go

```diff
@@ -32,20 +32,50 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// IOMode is the type of the mode used for I/O of detect and transform scripts.
+type IOMode string
+
+const (
+	envMode      IOMode = "env"
+	argMode      IOMode = "arg"
+	envDelimiter        = "="
+)
+
 // Executable implements transformer interface and is used to write simple external transformers
 type Executable struct {
 	Config     transformertypes.Transformer
 	Env        *environment.Environment
 	ExecConfig *ExecutableYamlConfig
 }
 
+// InputFormat input/output mode definition
+type InputFormat struct {
+	Mode    IOMode `yaml:"mode"`
+	EnvName string `yaml:"envName,omitempty`
+	Path    string
+}
+
+// OutputFormat input/output mode definition
+type OutputFormat struct {
+	Mode    IOMode `yaml:"mode"`
```

Comment:
Are the modes supported for input and output the same? For example, I am not sure how arg mode or even env mode will work for output